doc/how_to/migrate/from_streamlit.md

```diff
@@ -0,0 +1,950 @@
+    yield layout # This will display the layout
+
+    for i in range(0,10):
+        result = model()
+        message = f"Result {i}: {result}"
+        layout.append(message)
+        yield layout # This will force the UI to update
+
+    layout.pop(0)
+    run_input.disabled=False
+    yield layout # This will force the UI to update
+
+
+pn.Column(run_input, pn.bind(results, run_input)).servable()
+```
+
+![Panel Multiple Results Example](https://user-images.githubusercontent.com/42288570/243421842-b6c29bb0-b814-4d96-9918-946deea807ff.gif)
+
+### Interactivity Migration Steps
+
+You should
+
+- Move your business logic to functions. Business logic can be code to load data,
+create plots, calculate the mass of the milky way, train models, do inference etc.
+- Use `pn.bind` to specify which functions are bound to which widgets.
+- Use generator functions (`yield`) if you want to update the UI multiple times during the
+  code execution.
+- Indicate activity if needed. You can use one of
+[Panels Indicators](https://panel.holoviz.org/reference/index.html#indicators) and/ or
+the `.disabled`+`.loading` parameter on Panels components.
+
+## Caching
+
+One of the key concepts in Streamlit is *caching*. In Streamlit your entire script is rerun
+*top to bottom* on user interactions. Without caching you would be
+reconnecting to your database, reloading your dataset and rerunning your expensive calculation
+every time a user clicks a `Button` or changes a `slider` value. This would make your application
+very slow.
+
+In Panel only your *bound functions* are rerun on user interactions.
+
+In Panel you would use the
+
+- *session cache* (`pn.cache`) to speed up expensive, bound functions for a user session
+- *global cache* (`pn.state.as_cached`) to speed up expensive, bound functions or to share
+Python objects globally. I.e. across user sessions.
+
+Check out the [Cache How-To Guides](https://panel.holoviz.org/how_to/caching/index.html) for the
+details.
+
+### Session Cache Example
+
+This session will show you how to cache for a user session.
+
+#### Streamlit Session Cache Example
+
+```python
+from time import sleep
+
+import numpy as np
+import streamlit as st
+from matplotlib.figure import Figure
+
+@st.cache_data
+def get_data():
+    print("get_data func")
+    sleep(1.0)
+    return np.random.normal(1, 1, size=100)
+
+@st.cache_data(hash_funcs={Figure: lambda _: None})
+def plot(data, bins):
+    print("plot func", bins)
+    sleep(0.25)
+    fig = Figure(figsize=(8,4))
+    ax = fig.subplots()
+    ax.hist(data, bins=bins)
+    return fig
+
+data = get_data()
+bins = st.slider(value=20, min_value=10, max_value=30, step=1, label="Bins")
+st.pyplot(plot(data, bins))
+```
+
+I've added
+
+- `sleep` statements to make the functions more *expensive*.
+- `print` statements to show you that the functions are only runs once for a specific set of
+arguments.
+
+#### Panel Session Cache Example
+
+```python
+from time import sleep
```

Comment:
I'm not sure this is working as expected. I made the sleep 10 seconds and reran with cache; it's still 10 seconds.

And I'm not sure if caching matplotlib figures is a good example (if it's even possible); usually it's for fetching big data.

Reply:
The caching example works for me. 

In the video below I changed the `plot` func to sleep 2 secs and show a loading_indicator. You can clearly see it works (for me 😄 )

https://github.com/holoviz/panel/assets/42288570/82677e8b-ab9d-49ad-ac8c-61d94ecec9bc

I believe that caching is not only for data. I can be for storing any object that is expansive to create and used often. Also Matplotlib figures.